### ISSUE TITLE
Autohinting

### ISSUE THREAD

**Issue by arrowtype** [arrowtype/name-sans-roadmap#43]
TL;DR: I’ve mostly solved this in the build with `source/03-build-scripts/helpers/set-postscript-hinting-info.py`, because [FontMake doesn’t currently interpolate blueZones](https://github.com/googlefonts/fontmake/issues/824) during UFO interpolation.

## OTFs

I’ve tested this in InDesign and in a PDF, for Text Regular, Standard Bold, and Display Hairline. It makes a noticeable improvement to text at smaller sizes, especially around 8pt and below.

Set this up in the main build:
- [x] Set UFO Postscript Hinting info in the build prep
  - [x] Use baseline, x-height, cap height, ascender, bottom of `dotcomb`, and top of `four.onum` to set BlueValues
  - [x] Use descender, bottom of `g` (or maybe not?), and bottom of `four.onum` to set OtherBlues
  - [ ] (also correct blueValues and OtherBlues for primary sources)
  - [x] Set `StemSnapH` and `StemSnapV` ... somehow. Try:
    - [x] measure width of `idotless` and `l.text` for `StemSnapH`
    - [x] measure height of `hyphen` and vertical distance between rightmost coordinates (hook) of `l.default` for `StemSnapV`
- [x] add psautohint step to OTF build loop. Make sure to exclude `.notdef`

Later
- [ ] check status of CFF2 ... can these be autohinted?

## TTFs
- [ ] just try running ttfautohint for woff2 files
- [ ] make (or find?) web waterfall test for several styles
- [ ] test on windows & android

**Comment by arrowtype** [comment]
The `ij` dots are sometimes fusing with their bases, at small sizes (around 8px), in a way they don’t without hinting. This is because at certain sizes, the xheight is basically enlarged, which pulls the stem into the dot. I have tested adding a `BlueZone` for the bottom of the dots, but that doesn’t fix it.

Something I haven’t tested: positioning the `idotless` just _below_ the x-height, in order to "dodge" the hinting. This could potentially work well if only applied to Text sizes, and not to Display sizes. (And if it is a problem in Standard, I might add a support glyph at ops 24).

**Comment by arrowtype** [comment]
Test: positioning all overshoots to +/- 20 units of metrics lines. Doing this in the `harmonized-overshoots` branch, in core alphabets and figures (but not yet in absolutely every glyph).

- [ ] build, with name suffix
- [ ] test rendering after PSautohint
  - [ ] InDesign, vs non-harmonized
  - [ ] InDesign to PDF
  - [ ] InDesign to JPEG
- [ ] Windows, woff2 with ttfautohint, on cyreal test page

**Comment by arrowtype** [comment]
The test above made no perceptible difference, but I mostly prefer the changes made for it. I’m keeping it!
